prog/dftb+/lib_common/mpienv.F90

```diff
@@ -0,0 +1,95 @@
+!--------------------------------------------------------------------------------------------------!
+!  DFTB+: general package for performing fast atomistic simulations                                !
+!  Copyright (C) 2017  DFTB+ developers group                                                      !
+!                                                                                                  !
+!  See the LICENSE file for terms of usage and distribution.                                       !
+!--------------------------------------------------------------------------------------------------!
+
+!> Contains MPI related einvironment settings
+module mpienv
+  use mpifx
+  use message
+  implicit none
+  private
+
+  public :: TMpiEnv, TMpiEnv_init
+
+
+  !> Contains MPI related einvironment settings
+  type :: TMpiEnv
+
+    !> Global MPI communicator
+    type(mpifx_comm) :: globalComm
+
+    !> Communicator to access processes within current group
+    type(mpifx_comm) :: groupComm
+
+    !> Communicator to access equivalent processes in other groups
+    type(mpifx_comm) :: interGroupComm
+
+    !> Size of the process groups
+    integer :: groupSize
+
+    !> Number of processor groups
+    integer :: nGroup
+
+    !> Group index of the current process (starts with 0)
+    integer :: myGroup
+
+    !> Global rank of the processes in the given group
+    integer, allocatable :: groupMembers(:)
+
+    !> Whether current process is the global master
+    logical :: tGlobalMaster
+
+    !> Whether current process is the group master
+    logical :: tGroupMaster
+
+  end type TMpiEnv
+
+
+contains
+
+  !> Initializes MPI environment.
+  subroutine TMpiEnv_init(this, nGroup)
+
+    !> Initialised instance on exit
+    type(TMpiEnv), intent(out) :: this
+
+    !> Number of process groups to create
+    integer, intent(in) :: nGroup
+
+    character(100) :: tmpStr
```

Comment:
why 100 instead of one of the standard string lengths elsewhere in the accuracy module?

Reply:
Done.

However, probably for local variables (which received or passed via subroutine calls) we could set the lengths arbitrary...